python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -746,13 +761,17 @@ def summary(self):
         pending_nodes = []
         failed_nodes = []
 
+        non_failed = set()
+
         for node_id in all_node_ids:
             ip = self.provider.internal_ip(node_id)
             node_tags = self.provider.node_tags(node_id)
             if node_tags[TAG_RAY_NODE_KIND] == NODE_KIND_UNMANAGED:
                 continue
             node_type = node_tags[TAG_RAY_USER_NODE_TYPE]
 
+            failed = False
```

Comment:
can we remove this and do non_failed.add(node_id) in the if is_active and in the if is_pending (makes it easier to read, one line shorter as well)?